File: generate-icons.js
// Simple Node.js script to generate PWA icons from SVG
// This requires Node.js and sharp library
// Run with: npm install sharp && node generate-icons.js

const sharp = require('sharp');
const fs = require('fs');

async function generateIcons() {
    const svgBuffer = fs.readFileSync('icon.svg');
    const sizes = [72, 96, 128, 144, 152, 192, 384, 512];
    
    for (const size of sizes) {
        await sharp(svgBuffer)
            .resize(size, size)
            .png()
            .toFile(`icons/icon-${size}x${size}.png`);
        console.log(`Generated icon-${size}x${size}.png`);
    }
    
    console.log('All icons generated successfully!');
}

// Check if running directly
if (require.main === module) {
    generateIcons().catch(console.error);
}

module.exports = { generateIcons };

File: generate_icons.py
#!/usr/bin/env python3
"""
Simple Python script to generate PWA icons from SVG
Requires: pip install cairosvg pillow
"""

import os
from PIL import Image
import cairosvg

def svg_to_png(svg_path, png_path, size):
    """Convert SVG to PNG with specified size"""
    cairosvg.svg2png(
        url=svg_path,
        write_to=png_path,
        output_width=size,
        output_height=size
    )
    print(f"Generated {png_path}")

def main():
    # Create icons directory if it doesn't exist
    os.makedirs('icons', exist_ok=True)
    
    # Generate required icon sizes
    sizes = [192, 512]
    
    for size in sizes:
        png_path = f'icons/icon-{size}x{size}.png'
        try:
            svg_to_png('icon.svg', png_path, size)
        except Exception as e:
            print(f"Error generating {png_path}: {e}")
            # Create a simple placeholder image
            img = Image.new('RGB', (size, size), color='#16213e')
            img.save(png_path)
            print(f"Created placeholder {png_path}")

if __name__ == '__main__':
    main()
    print("Icon generation complete!")